src/use-swr.ts

```diff
@@ -542,6 +547,15 @@ export const useSWRHandler = <Data = any, Error = any>(
   // If there is no `error`, the `revalidation` promise needs to be thrown to
   // the suspense boundary.
   if (suspense && isUndefined(data) && key) {
+    // SWR should throw when trying to use Suspense on the server with React 18,
+    // without providing any initial data. See:
+    // https://github.com/vercel/swr/issues/1832
+    if (!IS_REACT_LEGACY && IS_SERVER) {
```

Comment:
should we throw whatever data and key is?

Reply:
@huozhi As the comment and discussions mentioned, we should only throw **if we are going to** suspend on the server side.